File: Thread/Pre/EOCOperationClassONe/EOCOperationClassPer/ViewController.m
//
//  ViewController.m
//  EOCOperationClassPer
//
//  Created by EOC on 2017/4/14.
//  Copyright © 2017年 EOC. All rights reserved.
//

#import "ViewController.h"
#import "EocOperation.h"
#import "EOCBlockOperationVC.h"
#import "EOCInvocationOperationVC.h"
#import "EOCOpererationVC.h"
#import "EOCOperationComplexVC.h"
#import "ThreadDeadViewCtr.h"
#import "ThreadUseViewCtr.h"


@interface ViewController ()<UITableViewDelegate, UITableViewDataSource>{
    

}

@end

@implementation ViewController


- (void)viewDidLoad {
    [super viewDidLoad];
}

#pragma mark - tableView delegate
- (NSInteger)tableView:(UITableView *)tableView numberOfRowsInSection:(NSInteger)section{
    
    return 6;
}

- (CGFloat)tableView:(UITableView *)tableView heightForRowAtIndexPath:(NSIndexPath *)indexPath{
    
    return 50;
}

- (UITableViewCell *)tableView:(UITableView *)tableView cellForRowAtIndexPath:(NSIndexPath *)indexPath{
    
    UITableViewCell *cell = [tableView dequeueReusableCellWithIdentifier:@"cell"];
    if (!cell) {
        cell = [[UITableViewCell alloc] initWithStyle:UITableViewCellStyleDefault reuseIdentifier:@"cell"];
    }
    if (indexPath.row == 0) {
        cell.textLabel.text = @"NSThread";
    }
    else if (indexPath.row == 1) {
        cell.textLabel.text = @"系统：BlockOperation";
    }else if(indexPath.row == 2){
        cell.textLabel.text = @"系统：InvocationOperation";
    }else if(indexPath.row == 3){
        cell.textLabel.text = @"自定义Operation对象";
    }else if(indexPath.row == 4){
        cell.textLabel.text = @"依赖";
    }else if(indexPath.row == 5){
        cell.textLabel.text = @"僵死线程";
    }
    return cell;
    
}

- (void)tableView:(UITableView *)tableView didSelectRowAtIndexPath:(NSIndexPath *)indexPath{
    
    
    UIViewController *viewCtr = nil;
    if (indexPath.row == 0){
        viewCtr = [ThreadUseViewCtr new];
    }
    else if (indexPath.row == 1) {
        viewCtr = [EOCBlockOperationVC new];
    }else if(indexPath.row == 2){
        viewCtr = [EOCInvocationOperationVC new];
    }else if(indexPath.row == 3){
        viewCtr = [EOCOpererationVC new];
    }else if(indexPath.row == 4){
        viewCtr = [EOCOperationComplexVC new];
    }else if(indexPath.row == 5){
        viewCtr = [ThreadDeadViewCtr new];
    }
    
    [self.navigationController pushViewController:viewCtr animated:YES];
}


@end

File: EventThoery/EventThoery/ViewController.swift
//
//  ViewController.swift
//  EventThoery
//
//  Created by Casey on 02/05/2019.
//  Copyright © 2019 nb. All rights reserved.
//

import UIKit

class ViewController: UIViewController, UITableViewDelegate, UITableViewDataSource {

    override func viewDidLoad() {
        super.viewDidLoad()
        self.title = ""
    }
    
    
    //MARK: tableview delegate
    let titleArr = ["Event Thoery", "Gesture", "Button", "Button And Gesture", "Scrollview", "Gesture Conflict", "Common Exist", "Frame delegate"]
    func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
        return titleArr.count
    }
    
    func tableView(_ tableView: UITableView, heightForRowAt indexPath: IndexPath) -> CGFloat {
        
        return 64
    }
    
    func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
        
        
        var cell = tableView.dequeueReusableCell(withIdentifier: "cell")
        if cell == nil {
            cell = UITableViewCell.init(style: .default, reuseIdentifier: "cell")
        }
        cell?.textLabel?.text = titleArr[indexPath.row]
        return cell!
    }
    
    func tableView(_ tableView: UITableView, didSelectRowAt indexPath: IndexPath) {
        tableView.deselectRow(at: indexPath, animated: true)
        
        if indexPath.row == 0 {
            
            self.navigationController?.pushViewController(EventTheoryVC(), animated: true)
            
        }else if indexPath.row == 1 {
            
            self.navigationController?.pushViewController(GestureViewCtr(), animated: true)
            
        }else if indexPath.row == 2 {
            
            self.navigationController?.pushViewController(ButtonViewCtr(), animated: true)
        }else if indexPath.row == 3 {
            
            self.navigationController?.pushViewController(ButtonAndGestureViewCtr(), animated: true)
        }else if indexPath.row == 4 {
            
            self.navigationController?.pushViewController(ScrollerMenuViewCtr(), animated: true)
        }else if indexPath.row == 5 {
            
            self.navigationController?.pushViewController(ConflictViewCtr(), animated: true)
        }else if indexPath.row == 6 {
            
            self.navigationController?.pushViewController(GestureCoexistViewCtr(), animated: true)
            
        }else if indexPath.row == 7 {
            
            self.navigationController?.pushViewController(FrameViewCtr(), animated: true)
            
        }
        
    }

    
}

